tiledb/sm/cpp_api/query_experimental.h

```diff
@@ -78,7 +81,162 @@ class QueryExperimental {
         ctx.ptr().get(), query.ptr().get(), &relevant_fragment_num));
     return relevant_fragment_num;
   }
+
+  /**
+   * Sets the data for a fixed/var-sized attribute/dimension.
+   *
+   * **Example:**
+   * @code{.cpp}
+   * tiledb::Context ctx;
+   * tiledb::Array array(ctx, array_name, TILEDB_WRITE);
+   * std::vector<int> data_a1 = {0, 1, 2, 3};
+   * Query query(ctx, array);
+   * query.set_data_buffer("a1", data_a1);
+   * @endcode
+   *
+   * @tparam T Attribute/Dimension value type
+   * @param name Attribute/Dimension name
+   * @param buf Buffer vector with elements of the attribute/dimension type.
+   **/
+  template <typename T>
+  static Query& set_data_buffer(
+      Query& query, const std::string& name, std::vector<T>& buf) {
+    // Checks
+    if (name == "__coords") {
```

Comment:
Do we still need that?

Reply:
I might be wrong, but I believe we need it for backward compatibility.

Reply:
Seems fine, ensures this namespaced variant will still work if someone uses it unconditionally for convenience. Marking resolved.